teraserver/python/tests/modules/FlaskModule/API/user/test_UserQueryDeviceTypes.py

```diff
@@ -0,0 +1,183 @@
+
+    def test_query_get_as_admin(self):
+        params = {'id_device_type': 0, 'list': False}
+        response = self._request_with_http_auth(username='admin', password='admin', payload=params)
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.headers['Content-Type'], 'application/json')
+        json_data = response.json()
+
+        params = {'id_device_type': 0, 'list': True}
+        response = self._request_with_http_auth(username='admin', password='admin', payload=params)
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.headers['Content-Type'], 'application/json')
+        json_data = response.json()
+        self.assertEqual(len(json_data), 4)
+
+        for i in range(3):
+            params = {'id_device_type': i+1, 'list': False}
+            response = self._request_with_http_auth(username='admin', password='admin', payload=params)
+            self.assertEqual(response.status_code, 200)
+            self.assertEqual(response.headers['Content-Type'], 'application/json')
+            json_data = response.json()
+            self.assertEqual(len(json_data), 1)
+
+            params = {'id_device_type': i+1, 'list': True}
+            response = self._request_with_http_auth(username='admin', password='admin', payload=params)
+            self.assertEqual(response.status_code, 200)
+            self.assertEqual(response.headers['Content-Type'], 'application/json')
+            json_data = response.json()
+            self.assertEqual(len(json_data), 4)
+
+        params = {'id_device_type': 5, 'list': False}
+        response = self._request_with_http_auth(username='admin', password='admin', payload=params)
+        self.assertEqual(response.status_code, 403)
+
+    def test_post_as_admin(self):
+        # This test should be run on a clean server (i.e. with only the defaults created)
+        # Else, the ID of the new devices created wont be aligned and the tests will fail
+
+        params = {'device_type': {'device_type_name': 'New_Device_Type',
+                                  'id_device_type': 0,
+                                  'device_type_key': 'new_device'}}
+        response = self._post_with_http_auth(username='admin', password='admin', payload=params)
+        self.assertEqual(response.status_code, 200)
+
+        # Create another instance of the same object - Fail Expected
+        response = self._post_with_http_auth(username='admin', password='admin', payload=params)
+        self.assertEqual(response.status_code, 501)
+
+        # Create an instance of the same key - Fail expected
+        params = {'device_type': {'device_type_name': 'New_Device_Type_2',
+                                  'id_device_type': 0,
+                                  'device_type_key': 'new_device'}}
+        response = self._post_with_http_auth(username='admin', password='admin', payload=params)
+        self.assertEqual(response.status_code, 501)
+
```

Comment:
Add a test where a non-superadmin user tries to create a device type - should fail.